src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {
+  FC,
+  CSSProperties,
+  useState,
+  Children,
+  ReactElement,
+  cloneElement,
+  ReactNode,
+} from 'react';
+import { classNames } from '../../utils/classNames';
+import DropdownOption from './DropdownOption';
+
+export interface IDropdownProps {
+  /** children must be React Element */
+  children?: ReactElement | ReactElement[];
+  /** set customized css class */
+  className?: string;
+  /** set dropDown to be disabled */
+  disabled?: boolean;
+  /** set customized css style */
+  cssStyle?: CSSProperties;
+  /** set default string on active option */
+  placeholder?: string;
+  /** a callback to provide current value */
+  onChange?: (val: string) => void;
+}
+
+interface PatDropdownSubComponents {
+  Option: typeof DropdownOption;
+}
+
+/**
+ * A dropdown allows user to select from multiple actions.
+ *
+ * ```js
+ * import { Dropdown } from 'pat-ui'
+ * ```
+ */
+const Dropdown: FC<IDropdownProps> & PatDropdownSubComponents = (props) => {
+  const {
+    className,
+    children,
+    cssStyle,
+    placeholder,
+    onChange,
+    disabled,
+  } = props;
+
+  const [isOptionListOpen, setIsOptionListOpen] = useState(false);
+  const [activeOption, setActiveOption] = useState(placeholder as ReactNode);
+
+  const toggleOptionList = () => {
+    setIsOptionListOpen(!isOptionListOpen);
+  };
+
+  const closeOptionList = () => {
+    if (isOptionListOpen) {
+      setIsOptionListOpen(!isOptionListOpen);
+    }
+  };
+
+  const setSelected = (val: string, innerChild: ReactNode) => {
+    // trigger callback function
+    if (onChange) {
+      onChange(val);
+    }
+
+    setActiveOption(innerChild);
+  };
+
+  let classStyles = classNames('dropdown');
+  if (isOptionListOpen) {
+    classStyles = classNames('dropdown', 'open');
+  }
+
+  return (
+    <div
+      className={
+        className ? `dropdown__wrapper ${className}` : 'dropdown__wrapper'
+      }
+    >
+      <div
+        className={classStyles}
+        style={cssStyle}
```

Comment:
logic is weird here, className is for wrapper and cssStyle is for the select. you need more doc to present this.

Reply:
cssStyle should be inside the wrapper class, I accidentally messed that one up. I also think classStyle is not a good name for what I'm trying to do. I will fix those two issues in the following commit.